internal/osbuild/tree_input.go

```diff
@@ -19,7 +19,7 @@ func NewTreeInput(reference string) *TreeInput {
 	return input
 }
 
-type PipelineTreeInputs map[string]TreeInput
+type PipelineTreeInputs map[string]Input
```

Comment:
IMO we should not start using the `Input` interface, but should instead converge the tree input implementation to the equivalent of changes done for files input https://github.com/osbuild/osbuild-composer/blob/main/internal/osbuild/files_input.go.